common/lib/common-utils/src/hashFileBrowser.ts

```diff
@@ -5,20 +5,31 @@
 
 import { IsoBuffer } from "./bufferBrowser";
 
+let insecureContextHashFn: ((f: IsoBuffer) => Promise<string>) | undefined;
+
+/**
+ * Set a hashing function to be called in place of hashFile's internal
+ * implementation when running under insecure contexts.  Not needed
+ * when running under Node.
+ * @param hashFn - The function that should be used in place of hashFile
+ */
+export function setInsecureContextHashFn(hashFn: (f: IsoBuffer) => Promise<string>) {
+    insecureContextHashFn = hashFn;
+}
+
 /**
  * Hash a file. Consistent within a session, but should not be persisted and
  * is not consistent with git.
- * Must be called under secure context for browsers
+ * If called under an insecure context for a browser, an override function
+ * needs to be set using setInsecureContextHashFn
  *
  * @param file - The contents of the file in a buffer
  * @returns The hash of the content of the buffer
  */
 export async function hashFile(file: IsoBuffer): Promise<string> {
-    // Handle insecure contexts (e.g. running with local services)
-    // by deferring to Node version, which uses a hash polyfill
-    const subtle = crypto.subtle;
-    if (subtle === undefined) {
-        return import("./hashFileNode").then(async (m) => m.hashFile(file));
+    // Use the function override if provided
+    if (insecureContextHashFn !== undefined) {
```

Comment:
Should we use this insecureContextHashFn only in insecure contexts? Or just always use it when provided? If it's always used, maybe we should consider renaming it to hashFunctionOverride?

Reply:
i don't think anyone should be using this outside of the developer scenario we're trying to hit with insecure contexts. i added an assert for such to enforce that

Reply:
Do we need to provide an error message in this assert here?

Reply:
Not sure the developer will have enough context without a message

Reply:
i'm pretty sure i'm going to be that developer 🤔
i don't think it's really necessary here because it's documented in the comments and it increases the size just a bit, but i can add something if you think it's necessary.

Reply:
That's fair.  No message sounds reasonable until we get feedback that we need to do more :)